controllers/clusters/kafka_controller.go

```diff
@@ -382,6 +393,54 @@ func (r *KafkaReconciler) handleDeleteUser(
 	return nil
 }
 
+func (r *KafkaReconciler) detachUser(
+	ctx context.Context,
+	kafka *v1beta1.Kafka,
+	l logr.Logger,
+	userRef *v1beta1.UserReference) error {
+	req := types.NamespacedName{
+		Namespace: userRef.Namespace,
+		Name:      userRef.Name,
+	}
+
+	u := &kafkamanagementv1beta1.KafkaUser{}
+	err := r.Get(ctx, req, u)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			l.Info("Kafka user is not found", "request", req)
+			r.EventRecorder.Event(kafka, models.Warning, "Not Found",
+				"User is not found, please provide correct userRef.")
+
+			return err
+		}
+
+		l.Error(err, "Cannot get Kafka user", "user", u.Spec)
+		r.EventRecorder.Eventf(kafka, models.Warning, models.DeletionFailed,
+			"Cannot get Kafka user. user reference: %v", userRef)
+
+		return err
+	}
+	if _, exist := u.Status.ClustersEvents[kafka.Status.ID]; !exist {
```

Comment:
Add here new line and check similar cases and fix them if they exist.

Reply:
done